populate_analytics.js

```diff
@@ -0,0 +1,283 @@
+const mongoose = require('mongoose');
+const quickselect = require('quickselect');
+
+var carddb = require('./serverjs/cards.js');
+const Deck = require('./models/deck');
+const Cube = require('./models/cube');
+const Card = require('./models/card');
+const mongosecrets = require('../cubecobrasecrets/mongodb');
+
+const batch_size = 100;
+
+const basics = ['mountain','forest','plains','island','swamp'];
+
+const cardUses = {};
+const cardSizeUses = {
+    size180: {},
+    size360: {},
+    size450: {},
+    size540: {},
+    size720: {},
+    pauper: {},
+    legacy: {},
+    modern: {},
+    standard: {},
+    vintage: {},
+};
+
+//global cube stats
+const cubeCounts = {
+    total: 0,
+    size180: 0,
+    size360: 0,
+    size450: 0,
+    size540: 0,
+    size720: 0,
+    pauper: 0,
+    legacy: 0,
+    modern: 0,
+    standard: 0,
+    vintage: 0,
+}
+
+const correlationIndex = {};
+const correlations = [];
+
+//use correlationIndex for index
+const cubesWithCard = [];
+
+function createCorrelations() {
+    const totalCards = carddb.cardnames.length;
+    for (var i = 0; i < totalCards; i ++) {
+        correlationIndex[carddb.cardnames[i].toLowerCase()] = i;
+        correlations.push([]);
+        cubesWithCard.push([]);
+        for (var j = 0; j < totalCards; j ++) {
+            correlations[i].push(0);
+        }
+        if((i+1)%100==0){
+            console.log('Finished: ' + (i+1) + ' of ' + totalCards + ' correlations.');
+        }
+    }
+    console.log('Finish init of correlation matrix.')
+}
+
+function attemptIncrement(obj, propname) {
+    if(!obj[propname]) {
+        obj[propname] = 0;
+    }
+    obj[propname]++; 
+
+}
+
+async function processDeck(deck) {
+    if(deck.playerdeck && deck.playerdeck.length > 0) {
+        //flatten array
+        const deckCards = [];
+        deck.playerdeck.forEach(function(col, index) {
+            if(Array.isArray(col)) {
+                col.forEach(function(row, index2) {
+                    deckCards.push(row.details.name.toLowerCase().normalize('NFD')
```

Comment:
Is there a reason not to use `carddb.normalizedName(row.details)` here? (And below with `col.details`?)